lib/SILOptimizer/Mandatory/PMOMemoryUseCollector.cpp

```diff
@@ -555,8 +555,28 @@ bool ElementUseCollector::collectUses(SILValue Pointer, unsigned BaseEltNo) {
         continue;
 
       // If this is an @inout parameter, it is like both a load and store.
-      case ParameterConvention::Indirect_Inout:
       case ParameterConvention::Indirect_InoutAliasable: {
+        // FIXME: The @inout_aliasable convention is used for indirect captures
+        // of both 'let' and 'var' variables. Using a more specific convention
+        // for 'let' properties like @in_guaranteed unfortunately exposes bugs
+        // elsewhere in the pipeline. A 'let' capture cannot really be mutated
+        // by the callee, and this is enforced by sema, so we can consider it
+        // a nonmutating use.
+        bool isLet = true;
+        
+        for (unsigned i = 0; i < TheMemory.NumElements; ++i) {
```

Comment:
Same here.

Reply:
Also, please add a test case for PMO.

Reply:
What is PMO?

Reply:
PMO == Predictable Memory Opts. Predictable Mem Opts and DI now have two different use collector impls. So if you only care about DI, just change the Ownership one. I am waiting for you to finish this to rename the Ownership version to not have the Ownership suffix.

If your thing doesn't care about predictable memory opts, then I wouldn't change this.

Reply:
You may want to wait to change the name until we wrap up the 4.2 branch, since that will make cherry-picking changes over more difficult and error-prone. Does PMO affect diagnostics?

Reply:
Furthermore, does the formerly-DIMemoryUseCollector still have a role in DI on the 4.2 branch?

Reply:
No PMO does not. It is only used for predictable memory operations. The idea was to split the two passes so that they could evolve independently. They had a bunch of non-shared code that made it hard to reason about.

I think that git is good enough to handle moves... but I can wait.

Reply:
My concern with the move here is that you'd be replacing a file with a different-but-similar-enough file that Git might mistake it for an in-place modification rather than a sequence of renames.